src/targeting/content.ts

```diff
@@ -0,0 +1,245 @@
+	'400+ page view per minute': '1',
+} as const;
+
+const platform = {
+	R2: 'r2',
+	NextGen: 'ng',
+	MobileApp: 'app',
+	AcceleratedMobilePages: 'amp',
+} as const;
+
+const contentType = [
+	'article',
+	'audio',
+	'crossword',
+	'gallery',
+	'interactive',
+	'liveblog',
+	'network-front',
+	'section',
+	'tag',
+	'video',
+] as const;
+
+/**
+ * Content Targeting comes from the server
+ *
+ * For a specific URL, it will only change on
+ * - a Composer/CAPI update
+ * - a rendering platform capability update
+ * - a main media update
+ * - a series tag update
+ * - a surge in page views per minute
+ *
+ */
+export type ContentTargeting = {
+	/**
+	 * **Bl**og tags – [see on Ad Manager][gam]
+	 *
+	 * Type: _Dynamic_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=186687
+	 */
+	bl: string[];
+
+	/**
+	 * **Br**anding - [see on Ad Manager][gam]
+	 *
+	 * Type: _Predefined_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=259767
+	 */
+	br: typeof branding[keyof typeof branding] | null;
+
+	/**
+	 * **Co**ntributor - [see on Ad Manager][gam]
+	 *
+	 * Array of all contributors to the content on the page
+	 *
+	 * Type: _Dynamic_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=186207
+	 */
+	co: string[];
+
+	/**
+	 * **C**ontent **T**ype - [see on Ad Manager][gam]
+	 *
+	 * Type: _Predefined_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=177807
+	 */
+	ct: typeof contentType[number];
+
+	/**
+	 * **D**ot**c**om-**r**endering **E**ligible - [see on Ad Manager][gam]
+	 *
+	 * Type: _Predefined_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=177807
+	 */
+	dcre: True | False;
+
+	/**
+	 * **Edition** - [see on Ad Manager][gam]
+	 *
+	 * Type: _Predefined_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=174207
+	 */
+	edition: typeof editions[keyof typeof editions];
+
+	/**
+	 * **K**eywords - [see on Ad Manager][gam]
+	 *
+	 * Type: _Dynamic_
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=177687
+	 */
+	k: string[];
+
+	/**
+	 * **Ob**server Content - [see on Ad Manager][gam]
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=256887
+	 */
+	ob: 't' | null;
+
+	/**
+	 * **P**latform - [see on Ad Manager][gam]
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=180207
+	 */
+	p: typeof platform[keyof typeof platform];
+
+	/**
+	 * Rendering Platform - [see on Ad Manager][gam]
+	 *
+	 * [gam]: https://admanager.google.com/59666047#inventory/custom_targeting/detail/custom_key_id=11881005
+	 */
+	rp: 'dotcom-rendering' | 'dotcom-platform';
```

Comment:
@Amouzle could this value be embedded inside of `p` (Platform)?

Reply:
I guess currently `p` is generated by [`commercial-shared`](https://github.com/guardian/commercial-shared/blob/master/src/main/scala/com/gu/commercial/display/AdTargetParam.scala#L138-L146).